### ISSUE TITLE
Potential Bug in WAE-MMD loss function

### ISSUE THREAD

**Issue by tomastokar** [clementchadebec/benchmark_VAE#156]
In `wae-mmd` the `loss_function`  the argument `z_prior` is obtained from: 

```python
z_prior = torch.randn_like(z, device=x.device)
```

which yields random numbers from a uniform distribution on the interval [0, 1), but according to the paper (Tolstikhin at al.), page 5, algorithm 2, `z_prior` should be sampled from the prior distribution, which is, conventionally, Gaussian.

**Comment by clementchadebec** [comment]
Hello @tomastokar,

Thank you for opening this issue. Actually, unless I am mistaken, it seems correct since `torch.randn_like` samples from a Gaussian distribution. See the [pytorch doc](https://pytorch.org/docs/stable/generated/torch.randn_like.html).

I hope this helps.

**Comment by tomastokar** [comment]
Hi @clementchadebec , 

Oh, just realized there are two different functions with similar name `torch.rand_like` vs `torch.randn_like`, one samples from uniform, another from Gaussian, respectively. Thanks you very much for elucidation.

**Comment by tomastokar** [comment]
Closing, as this is not an issue
